### ISSUE TITLE
removing tracks at a certain position without passing a track id

### ISSUE THREAD

**Issue by kasperkamperman** [jwilsson/spotify-web-api-php#116]
I'd like to remove tracks if my playlist is too long. According to the Spotify API this should be doable to only pass a positions array. However I get the idea that this isn't implemented. Although it also could be that my syntax is wrong.

https://developer.spotify.com/web-api/remove-tracks-playlist/
Paragraph: Removing the track at a given position in a specific playlist snapshot

First try:
```
$tracks = [['positions' => 5]];
echo $api->deleteUserPlaylistTracks($me->id, $playlistId, $tracks);
```
This try will give the error:
Notice: Undefined index: id in /Applications/MAMP/htdocs/spotifyfm/src/SpotifyWebAPI.php on line 339 JSON body doesn't conform to specification

Second try:
```
$tracks = [
  ['id' => ''],
  ['positions' => 5]
];
```

This try will give the error:
Notice: Undefined index: id in /Applications/MAMP/htdocs/spotifyfm/src/SpotifyWebAPI.php on line 339
JSON body contains an invalid track uri: spotify:track:

**Comment by jwilsson** [comment]
Hi!
It looks like we missed that case in `deleteUserPlaylistTracks()`. Right now, the ID is always required but it should be made optional.

Would you be interested in submitting a PR with a fix?

**Comment by kasperkamperman** [comment]
I'm afraid I don't have enough knowledge of the inner works to fix it. 

I tried testing to comment the "id" lines in the deleteUserPlaylistTracks function, just to test. However I get the error "JSON body contains a track without URI". I also pass the snapshot_id which seems obligatory in this case. 

I think this error is thrown by Spotify, so I don't know what is wrong now. 

What I notice is that this specific implementation is not mentioned on the beta version of the developer site. So maybe this feature is removed? 
https://beta.developer.spotify.com/documentation/web-api/reference/playlists/remove-tracks-playlist/

```
public function deleteUserPlaylistTracks($userId, $playlistId, $tracks, $snapshotId = '') {
        $options = [];

        if ($snapshotId) {
            $options['snapshot_id'] = $snapshotId;
        }

        $options['tracks'] = array_map(function ($track) {
            $track = (array) $track;

            if (isset($track['positions'])) {
                $track['positions'] = (array) $track['positions'];
            }

            //$track['uri'] = $this->idToUri($track['id'], 'track');
            //unset($track['id']);

            return $track;
        }, $tracks);

```
The error:
```

```

**Comment by jwilsson** [comment]
@kasperkamperman No worries! I'll take a look at it, unless someone else beats me to it.

**Comment by kasperkamperman** [comment]
Thanks for the fix!

**Comment by jwilsson** [comment]
@kasperkamperman A fix for this has been released in `2.2.0`. Something like this should work:

```php
$tracks = [
  'positions' => [1, 2, 3],
];

$api->deleteUserPlaylistTracks($me->id, $playlistId, $tracks, $snapshotId);
```

Due note that the `$snapshotId` is required, but you can get it from a call to `getUserPlaylist()`.
